lib/foreman_maintain/cli/report_command.rb

```diff
@@ -3,23 +3,125 @@ module Cli
     class ReportCommand < Base
       extend Concerns::Finders
 
+      def generate_report
+        scenario = run_scenario(Scenarios::Report::Generate.new({}, [:reports])).first
+
+        # description can be used too
+        report_data = scenario.steps.map(&:data).compact.reduce(&:merge).transform_keys(&:to_s)
+        report_data['version'] = 1
+        report_data
+      end
+
+      def save_report(report, file)
+        if file
+          File.write(file, report)
+        else
+          puts report
+        end
+      end
+
       option '--output', 'FILE', 'Output the generate report into FILE'
       subcommand 'generate', 'Generates the usage reports' do
         def execute
-          scenario = run_scenario(Scenarios::Report::Generate.new({}, [:reports])).first
-
-          # description can be used too
-          report_data = scenario.steps.map(&:data).compact.reduce(&:merge).transform_keys(&:to_s)
-          report_data['version'] = 1
+          report_data = generate_report
           yaml = report_data.to_yaml
-          if @output
-            File.write(@output, yaml)
-          else
-            puts yaml
-          end
+          save_report(yaml, @output)
+
           exit runner.exit_code
         end
       end
+
+      option '--input', 'FILE', 'Input the report from FILE'
+      option '--output', 'FILE', 'Output the condense report into FILE'
+      option '--max-age', 'HOURS', 'Max age of the report in hours'
+      subcommand 'condense',
+        'Generate a JSON formatted report with condensed data from the original report.' do
+        def execute
+          data = if fresh_enough?(@input, @max_age)
+                   YAML.load_file(@input)
+                 else
+                   generate_report
+                 end
+
+          report = condense_report(data)
+          report = prefix_keys(report)
+          save_report(JSON.dump(report), @output)
+        end
+
+        def condense_report(data)
+          result = {}
+          %w[advisor_on_prem_remediations_count rhel_ai_workload_host_count].each do |key|
```

Comment:
Why is it needed? Both fields are using `sql_count`, which means the fields will always be in the report

Reply:
Back then when I wrote this it wasn't the case. I can change it, but on the other hand, does it hurt to be extra defensive and not assume anything?